src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala

```diff
@@ -348,31 +440,32 @@ trait GraphDAO {
   def getTypeParams(tpe: Type) = tpe.asInstanceOf[TypeRefApi].args
 
   //SAVE METHODS
-  def saveObject[T <: DomainObject :TypeTag :ClassTag](obj: T, vertex: Vertex, wsc: Option[WorkspaceContext], graph: Graph): Unit = {
-    saveObject(typeOf[T], obj, vertex, wsc, graph)
+  def saveObject[T <: DomainObject :CachedType](obj: T, vertex: Vertex, wsc: Option[WorkspaceContext], graph: Graph): Unit = {
+    saveObject(cachedTypeOf[T], obj, vertex, wsc, graph)
   }
 
-  def saveObject(tpe: Type, obj: DomainObject, vertex: Vertex, wsc: Option[WorkspaceContext], graph: Graph): Unit = {
+  def saveObject(tpe: CachedType[_], obj: DomainObject, vertex: Vertex, wsc: Option[WorkspaceContext], graph: Graph): Unit = {
     //Serialize out each of the case class properties.
     getPropertiesAndValues(tpe, obj).foreach({
       case (tp, prop, value) => saveProperty(tp, prop, value, vertex, wsc, graph)
     })
 
     if( wsc.isDefined ) {
-      saveProperty(typeOf[DateTime], "lastModified", DateTime.now, wsc.get.workspaceVertex, wsc, graph)
+      saveProperty(cachedTypeOf[DateTime], "lastModified", DateTime.now, wsc.get.workspaceVertex, wsc, graph)
     }
   }
 
   // a function to uniquely identify a graph object relative to siblings, by checking the idFields for equality
-  private def domainObjectFilterFn(tpe: Type, obj: DomainObject)(vertex: Vertex): Boolean = {
-    obj.idFields.forall { field => obj.getFieldValue(tpe, field) == vertex.getProperty(field) }
+  private def domainObjectFilterFn(tpe: CachedType[_], obj: DomainObject): Vertex => Boolean = {
+    val idFieldValues = obj.idFields.map(field => (field, obj.getFieldValue(tpe.tpe, field)))
```

Comment:
ah, i see. some commentary here might be good in order to clear this up, since it's confused two people
"this returns a filter function where the id field values have already been looked up, rather than calling `getFieldValue` every time the filter function is run"